### PR TITLE
List and start dynamic services

### PR DESCRIPTION
**Pull request by odeimaiz**

### CODE REVIEW

File: services/web/client/source/class/qxapp/components/workbench/SettingsView.js [hunk 1]
```diff
@@ -65,6 +65,16 @@ qx.Class.define("qxapp.components.workbench.SettingsView", {
     __applyNode: function(node, oldNode, propertyName) {
       this.__settingsBox.removeAll();
       this.__settingsBox.add(node.getPropsWidget());
+
+      // Show viewer
+      if (Object.prototype.hasOwnProperty.call(node.getMetadata(), "viewer")) {
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/components/workbench/SettingsView.js]: hmmm can't you just write:

```
if (node.getMetadata().viewer)
```

File: services/web/client/source/class/qxapp/components/workbench/Workbench.js [hunk 2]
```diff
@@ -366,26 +367,25 @@ qx.Class.define("qxapp.components.workbench.Workbench", {
       let nodeBase = new qxapp.components.workbench.NodeBase();
       nodeBase.setMetadata(nodeMetaData);
 
-      if (nodeBase.getNodeImageId() === "modeler") {
-        const slotName = "startModeler";
+      const imageId = nodeBase.getNodeImageId();
+      if (imageId.includes("dynamic")) {
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/components/workbench/Workbench.js]: can't you just write
```
if (imageId.dynamic)
```
?

File: services/web/client/source/class/qxapp/components/workbench/Workbench.js [hunk 3]
```diff
@@ -366,26 +367,25 @@ qx.Class.define("qxapp.components.workbench.Workbench", {
...
+        const slotName = "startDynamic";
         let socket = qxapp.wrappers.WebSocket.getInstance();
         socket.on(slotName, function(val) {
           if (val["service_uuid"] === nodeBase.getNodeId()) {
-            let portNumber = val["containers"][0]["published_ports"];
+            let portNumber = val["containers"][0]["published_ports"][0];
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/components/workbench/Workbench.js]: how about
```
let portNumber = val.containers[0].published_ports[0]
```

File: services/web/client/source/class/qxapp/components/workbench/Workbench.js [hunk 4]
```diff
@@ -366,26 +367,25 @@ qx.Class.define("qxapp.components.workbench.Workbench", {
...
             nodeBase.getMetadata().viewer.port = portNumber;
+            nodeBase.getMetadata().viewer.ip = "http://" + window.location.hostname;
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/components/workbench/Workbench.js]: hmmm ... this is NOT an ip address ...

File: services/web/client/source/class/qxapp/components/workbench/servicesCatalogue/ServicesCatalogue.js [hunk 5]
```diff
@@ -33,10 +33,13 @@ qx.Class.define("qxapp.components.workbench.servicesCatalogue.ServicesCatalogue"
     this.__allServices = qxapp.data.Fake.getFakeServices();
     let store = qxapp.data.Store.getInstance();
     store.addListener("servicesRegistered", function(e) {
-      console.log("servicesRegistered", e.getData());
       this.__addNewData(e.getData());
     }, this);
-    store.getServices();
+    store.getComputationalServices();
+    store.addListener("interactiveServicesRegistered", function(e) {
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/components/workbench/servicesCatalogue/ServicesCatalogue.js]: e => { ... });

File: services/web/client/source/class/qxapp/data/Converters.js [hunk 6]
```diff
@@ -21,6 +21,10 @@ qx.Class.define("qxapp.data.Converters", {
           metadata[field] = data[field];
         }
       });
+      // for dynamic services
+      if (Object.prototype.hasOwnProperty.call(data, "viewer")) {
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/data/Converters.js]: if (data.viewer){ ...
}

File: services/web/client/source/class/qxapp/data/Store.js [hunk 7]
```diff
@@ -4,32 +4,53 @@ qx.Class.define("qxapp.data.Store", {
...
           }
         }
         this.fireDataEvent("servicesRegistered", services);
       }, this);
       req.send();
+    },
+
+    getInteractiveServices: function() {
+      let socket = qxapp.wrappers.WebSocket.getInstance();
+      socket.removeSlot("getInteractiveServices");
```
Comment (oetiker) [inline services/web/client/source/class/qxapp/data/Store.js]: I would use REST for calls like this where there is no bidirectionality required ...

File: services/web/client/source/class/qxapp/components/workbench/Workbench.js (hunk 2 above)
Comment (odeimaiz) [reply services/web/client/source/class/qxapp/components/workbench/Workbench.js]: ImageID is a string

File: services/web/client/source/class/qxapp/data/Store.js (hunk 7 above)
Comment (odeimaiz) [reply services/web/client/source/class/qxapp/data/Store.js]: Agree, at some point next sprint, we will standarize the communication channels

**Pull request merged**